wrangler-core/src/main/java/io/cdap/directives/column/SetType.java

```diff
@@ -134,33 +134,35 @@ public Schema getOutputSchema(SchemaResolutionContext context) {
               if (field.getName().equals(col)) {
                 Integer outputScale = scale;
                 Integer outputPrecision = precision;
-                Schema fieldSchema = field.getSchema().getNonNullable();
-                Pair<Integer, Integer> scaleAndPrecision = getPrecisionAndScale(fieldSchema);
-                Integer inputSchemaScale = scaleAndPrecision.getSecond();
-                Integer inputSchemaPrecision = scaleAndPrecision.getFirst();
+                if (type.equalsIgnoreCase("decimal") && field.getSchema().isNullable()) {
+                  Schema fieldSchema = field.getSchema().getNonNullable();
```

Comment:
Also, I think this giant block that calculates precision and scale to use can be moved to a different function. Can refactor the function used in .map()

Reply:
Done